From 633d07a7b56dd583a32736d944df8b53a366acbf Mon Sep 17 00:00:00 2001
From: UtsavAgarwalADI <utsav.agarwal@analog.com>
Date: Fri, 13 Jun 2025 14:36:49 +0100
Subject: [PATCH] sc598-som: Adding support for revision E

- Updating qspi partitions to account for bigger flash storage
- Adding spi flash entry for chip compatibility
- Updating device tree to reflect new i2c gpio expander on the sc598-som

Signed-off-by: UtsavAgarwalADI <utsav.agarwal@analog.com>
---
 arch/arm64/boot/dts/adi/sc598-som.dtsi       | 80 +++++++++-----------
 arch/arm64/configs/sc598-som-ezkit_defconfig |  3 +-
 2 files changed, 38 insertions(+), 45 deletions(-)

diff --git a/arch/arm64/boot/dts/adi/sc598-som.dtsi b/arch/arm64/boot/dts/adi/sc598-som.dtsi
index 47060d5bce61..1af87078166e 100644
--- a/arch/arm64/boot/dts/adi/sc598-som.dtsi
+++ b/arch/arm64/boot/dts/adi/sc598-som.dtsi
@@ -204,10 +204,10 @@ &spi2 {
 	status = "okay";
 	cs-gpios = <&gpa 5 GPIO_ACTIVE_LOW>;
 
-	flash: is25lp512@0 {
+	flash: is25lp01g@0 {
 		#address-cells = <1>;
 		#size-cells = <1>;
-		compatible = "is25lp512", "jedec,spi-nor";
+		compatible = "is25lp01g", "jedec,spi-nor";
 		reg = <0>;
 		spi-cpha;
 		spi-cpol;
@@ -236,7 +236,7 @@ qspi_3: partition@3 {
 
 		qspi_4: partition@4 {
 			label = "JFFS2 Formatted RFS";
-			reg = <0x1020000 0x2FE0000>;
+			reg = <0x1020000 0x6FE0000>;
 		};
 
 	};
@@ -255,83 +255,75 @@ &i2c2 {
 	pinctrl-names = "default";
 	pinctrl-0 = <&i2c2_pins>;
 
-	ssw0: gpio@20 {
-		compatible = "microchip,mcp23018";
+	gpio@34 {
+		compatible = "adi,adp5587";
 		gpio-controller;
+		adi,gpio-only;
 		#gpio-cells = <2>;
-		reg = <0x20>;
+		reg = <0x34>;
 		status = "okay";
 
-		pinctrl-names = "default";
-		pinctrl-0 = <&ssw0pullups>;
-
-		ssw0pullups: pinmux {
-			bias-pull-up;
-			pins = "gpio0", "gpio1", "gpio2", "gpio3",
-				"gpio4", "gpio5", "gpio6", "gpio8", "gpio9";
-		};
-
-		led1 {
-			gpio-hog;
-			gpios = <0 0x0>;
-			output-high;
-			line-name = "led1-en";
-		};
-
-		led2 {
+		uart0 {
 			gpio-hog;
-			gpios = <1 0x0>;
-			output-high;
-			line-name = "led2-en";
+			gpios = <0 GPIO_ACTIVE_HIGH>;
+			output-low;
+			line-name = "~uart0-en";
 		};
 
-		led3 {
+		uart0-flow-en {
 			gpio-hog;
-			gpios = <2 0x0>;
+			gpios = <1 GPIO_ACTIVE_HIGH>;
 			output-high;
-			line-name = "led3-en";
+			line-name = "~uart0-flow-en";
 		};
 
 		spi2d2-d3 {
 			gpio-hog;
-			gpios = <3 0x0>;
+			gpios = <2 GPIO_ACTIVE_HIGH>;
 			output-low;
 			line-name = "~spi2d2-d3-en";
 		};
 
 		spi2flash-cs {
 			gpio-hog;
-			gpios = <4 0x0>;
+			gpios = <3 GPIO_ACTIVE_HIGH>;
 			output-low;
 			line-name = "~spi2flash-cs";
 		};
 
-		uart0 {
+		emmc {
 			gpio-hog;
-			gpios = <5 0x0>;
+			gpios = <8 GPIO_ACTIVE_HIGH>;
 			output-low;
-			line-name = "~uart0-en";
+			line-name = "emmc-en";
 		};
 
-		uart0-flow-en {
+		emmc-som-en {
 			gpio-hog;
-			gpios = <6 0x0>;
+			gpios = <9 GPIO_ACTIVE_HIGH>;
 			output-high;
-			line-name = "~uart0-flow-en";
+			line-name = "emmc-som-en";
 		};
 
-		emmc {
+		led1 {
 			gpio-hog;
-			gpios = <8 0x0>;
+			gpios = <15 GPIO_ACTIVE_HIGH>;
 			output-low;
-			line-name = "emmc-en";
+			line-name = "led1-en";
 		};
 
-		emmc-som-en {
+		led2 {
 			gpio-hog;
-			gpios = <9 0x0>;
-			output-high;
-			line-name = "emmc-som-en";
+			gpios = <16 GPIO_ACTIVE_HIGH>;
+			output-low;
+			line-name = "led2-en";
+		};
+
+		led3 {
+			gpio-hog;
+			gpios = <17 GPIO_ACTIVE_HIGH>;
+			output-low;
+			line-name = "led3-en";
 		};
 	};
 };
diff --git a/arch/arm64/configs/sc598-som-ezkit_defconfig b/arch/arm64/configs/sc598-som-ezkit_defconfig
index 041abcf35b61..28d4862d5eed 100644
--- a/arch/arm64/configs/sc598-som-ezkit_defconfig
+++ b/arch/arm64/configs/sc598-som-ezkit_defconfig
@@ -215,6 +215,7 @@ CONFIG_SPI_CADENCE_QUADSPI=y
 CONFIG_SPI_SPIDEV=y
 # CONFIG_PTP_1588_CLOCK_KVM is not set
 CONFIG_PINCTRL_MCP23S08=y
+CONFIG_KEYBOARD_ADP5588=y
 CONFIG_PINCTRL_SINGLE=y
 CONFIG_PINCTRL_ADSP_SC5XX=y
 CONFIG_SRUCTRL_ADSP_SC5XX=y
@@ -366,4 +367,4 @@ CONFIG_BLK_DEV_SD=y
 CONFIG_BLK_DEV_BSG=y
 CONFIG_USB_STORAGE=y
 CONFIG_SG_POOL=y
-CONFIG_RPMSG_CTRL=y
\ No newline at end of file
+CONFIG_RPMSG_CTRL=y
-- 
2.34.1

